gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -293,4 +293,93 @@ fluid.registerNamespace("gpii.transformer");
         return (60 * (first.hours - second.hours) + first.minutes - second.minutes) < 0;
     };
 
+    fluid.defaults("gpii.transformer.jsonArrayToTaggedArray", {
+        gradeNames: "fluid.standardTransformFunction"
+    });
+
+    /**
+     * Creates a tagged array from a json array, using the tag specified in the 'transformSpec'.
+     * If the values in the array are present in the 'elems' array of the 'transformSpec',
+     * the tag that will be added for that element will be the value specified in
+     * 'transformSpec.hasElemVal.isTrue', otherwise the value will be 'transformSpec.hasElemVal.isFalse'.
```

Comment:
This description isn't accurate, since the transform is hardwired to apply the special key "$t" which only has meaning in the special context of the GPII's XML to JSON converter. Firstly, parameterise the "$t" key into the transformSpec to make the transform properly general, and secondly, supply a full worked example in the comment so that it is clearer to readers what the transform really does.

Reply:
Updated with much more detailed doc.